Convert num_kernels to int64 before calling into CUDA GET_BLOCKS

soulitzer opened the pull request:
this addresses the example int64 issue in #44472.

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44688 Convert num_kernels to int64 before calling into CUDA GET_BLOCKS**

Differential Revision: [D23699819](https://our.internmc.facebook.com/intern/diff/D23699819)

soulitzer committed:
Convert num_kernels to int64 before calling into CUDA GET_BLOCKS

[ghstack-poisoned]

walterddr commented on aten/src/ATen/native/cuda/vol2col.cuh:
```diff
@@ -104,7 +104,7 @@ void vol2col(
   // We are going to launch channels * depth_col * height_col * width_col
   // kernels, each kernel responsible for copying a single-channel grid.
-  int num_kernels = channels * depth_col * height_col * width_col;
+  int64_t num_kernels = channels * depth_col * height_col * width_col;
```
I believe you need to cast one of the operands into int64_t first otherwise the overflowed int32 will be assigned to the `num_kernels` value.

also leaving variable type as `auto` is sufficient if you did the operand cast

soulitzer replied:
Ah, good catch. I've added the cast.

soulitzer committed:
Update on "Convert num_kernels to int64 before calling into CUDA GET_BLOCKS"

this addresses the example int64 issue in #44472.


Differential Revision: [D23699819](https://our.internmc.facebook.com/intern/diff/D23699819)

[ghstack-poisoned]
